src/nupic/algorithms/backtracking_tm_cpp.py

```diff
@@ -154,6 +163,44 @@ def __init__(self,
                             )
 
 
+  @classmethod
+  def getSchema(cls):
+    return BacktrackingTMCppProto
+
+
+  def write(self, proto):
+    super(BacktrackingTMCPP, self).write(proto.baseTM)
+    self.cells4.write(proto.cells4)
+    proto.makeCells4Ephemeral = self.makeCells4Ephemeral
+    proto.seed = self.seed
+    proto.checkSynapseConsistency = self.checkSynapseConsistency
+    proto.initArgs = json.dumps(self._initArgsDict)
+
+
+  @classmethod
+  def read(cls, proto):
+    # Use base class to create initial class
+    obj = BacktrackingTM.read(proto.baseTM)
+    obj.__class__ = cls
+
+    # Additional CPP-specific deserialization
+    newCells4 = Cells4.read(proto.cells4)
+    print newCells4
+    obj.cells4 = newCells4
+    obj.makeCells4Ephemeral = proto.makeCells4Ephemeral
+    obj.seed = proto.seed
+    obj.checkSynapseConsistency = proto.checkSynapseConsistency
+    obj._initArgsDict = json.loads(proto.initArgs)
+    obj._initArgsDict["outputType"] = str(obj._initArgsDict["outputType"])
```

Comment:
Is this casting away unicode? How about a short comment to that or other effect?  I think there may be an arg in `json.loads` that generates strings instead of unicode.